lib/stripe.rb

```diff
@@ -218,20 +225,6 @@ def self.max_network_retries=(val)
 
   private
 
-  def self.api_error(error, resp, error_obj)
-    APIError.new(error[:message], resp.code, resp.body, error_obj, resp.headers)
-  end
-
-  def self.authentication_error(error, resp, error_obj)
-    AuthenticationError.new(error[:message], resp.code, resp.body, error_obj,
-                            resp.headers)
-  end
-
-  def self.card_error(error, resp, error_obj)
-    CardError.new(error[:message], error[:param], error[:code],
-                  resp.code, resp.body, error_obj, resp.headers)
-  end
```

Comment:
These helper methods to create errors are a layer of obfuscation and don't add much, so I removed most of them (`general_api_error` survived because it's called more than once).